src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -510,6 +510,39 @@ internal void Assign(IEnumerable<TopicPartitionOffset> partitions)
         }
 
         /// <summary>
+        ///     Store offsets for one or more partitions.
+        ///   
+        ///     The offset will be committed (written) to the offset store according
+        ///     to `auto.commit.interval.ms` or manual offset-less commit().
+        /// </summary>
+        /// <remarks>
+        ///     `enable.auto.offset.store` must be set to "false" when using this API.
+        /// </remarks>
+        /// <param name="offsets">
+        ///     List of offsets to be commited.
+        /// </param>
+        /// <returns>
+        ///     Offset results with global or per-partition errors.
+        /// </returns>
+        internal OffsetResults StoreOffsets(IEnumerable<TopicPartitionOffset> offsets)
+        {
+            ThrowIfHandleClosed();
+            IntPtr cOffsets = GetCTopicPartitionList(offsets);
+            ErrorCode err = LibRdKafka.offsets_store(handle, cOffsets);
+            var result = GetTopicPartitionOffsetErrorList(cOffsets);
+            LibRdKafka.topic_partition_list_destroy(cOffsets);
+
+            if (err != ErrorCode.NoError)
+            {
+                return new OffsetResults(new Error(err));
```

Comment:
@edenhill  after looking at librdkafka code, it seems there is an incoherence between doc and implem

rdkafka.h
```
 * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success or an error code if
 *          none of the offsets could be stored.
```

rdkafka_offset.c
```
       return offsets->cnt > 0 && ok_cnt < offsets->cnt ?
              RD_KAFKA_RESP_ERR__UNKNOWN_PARTITION :
              RD_KAFKA_RESP_ERR_NO_ERROR;
```

We return unkown_partition if any offset could not be set because of unknown partition, even if other success.
I also wonder why there is no other possible error (transport or other)

On my previous comment in this case, this is more troublesome - we don't want to throw exception when it is not a complete failure, so either let it like this for now, either do some check on result (is there any correct offset stored?)

Also, this means the impl is not ok: in case a partition was not correctly set, we still want to create `offsetResults` from `result`, with corresponding error (so perhaps keeping a TopicPartitionOffsetError), to have both ok result and nok results.

Reply:
I'd say this is a bug in librdkafka. That should probably read `offsets->cnt > 0 && ok_cnt == 0`

Reply:
@treziac I agree that in case of global error, it makes sense to throw Exception. I am changing the global error checking to throw exception instead of returning OffsetResults with Error.

Reply:
@edenhill , will you be able to fix this bug you mentioned above and push it into the Nov 20 release?

Reply:
Yes, I will make it return UNKNOWN_PARTITION only if none of the partitions were found.

Reply:
and __INVALID_ARG if enable.auto.offset.reset is set to true (default)

Reply:
Please review: https://github.com/edenhill/librdkafka/pull/1523